api/service/version_endpoint_service.go

```diff
@@ -172,12 +185,13 @@ func (k *endpointService) DeployEndpoint(environment *models.Environment, model
 			}
 			endpoint.EnvVars = models.MergeEnvVars(pyfuncDefaultEnvVars, newEndpoint.EnvVars)
 		}
-	} else if model.Type == models.ModelTypeCustom {
+	} else {
 		endpoint.EnvVars = newEndpoint.EnvVars
 	}
 
-	originalEndpoint := *endpoint
-
+	endpoint.Status = models.EndpointPending
+	// Copy to avoid race condition
+	tobeDeployedEndpoint := *endpoint
 	endpoint.Status = models.EndpointPending
```

Comment:
why set this twice ya?

Reply:
I think it's typo when resolving some merge conflict

Reply:
Updated